content/docs/how-we-work/git/branch-merge.md

```diff
@@ -0,0 +1,50 @@
+---
+
+# Branching and Merging
+
+Git branching and merging are essential concepts for managing any software. By using branches, we can isolate different features or bug fixes, and merge them back into the main branch as soon as they are ready.
+
+## Branching for multiple features
+
+One of the most common uses for git branches is to manage multiple features. When we are working on a large project, it is often helpful to create a separate branch for each feature. This allows us to work on the feature independently without affecting each other. With regard to branch names, we may need to follow a specific nomenclature for branches or be project specific.
+
+To create a branch for a new feature, you can use the following command:
+
+```bash
+git branch my-new-feature
+```
+
+This will create a new branch named `my-new-feature`. You can then switch to the new branch by using the following command:
+
+```bash
+git checkout my-new-feature
+```
+
+Once you are on the new branch, you can start working on implementing the feature. When you are finished, you can push your changes and raise PR.
+
+## Resolving merge conflicts
+
+When you merge two branches, it is possible that there will be conflicts. This happens when the two branches make changes to the same file in the same proximity. In this case, Git may not be able to merge the branches automatically. You will need to resolve the conflicts manually.
+
+To resolve a merge conflict, you will need to open the file that has the conflict. You will then see two versions of the change, one from each branch. You will need to manually select the change that you want to keep. In case, one is not sure what changes to keep or what to remove, based on file commit history, reach out to the team member who last worked on that file and discuss and finalize the change.
+
+Once you have resolved the conflicts, you can save the file and commit the changes. Git will then merge the branches.
+
+## Merging Conflict on composer.lock
+
+Merging composer.lock is a common issue because it is a file that contains a list of all of the dependencies that are required by your project, as well as the specific versions of those dependencies. When we merge two branches, it is possible that the two branches have different versions of the dependencies. This can cause conflicts when we try to merge the composer.lock files.
+
+Generally, this conflict happens when people add/remove/update packages in different branches.
+
+The best way to handle this is to avoid having multiple people do it in the first place. Only one branch should deal with updates, and it should be merged quickly. And if there are other features that require a new module, they should add that module in a separate branch and merge it as soon as possible. This ensures that the composer.lock file rarely changes in individual branches.
```

Comment:
@Yemaneberhan-Lemma instead of saying below

`best way to handle this is to avoid having multiple people do it in the first place.
`
can we say

best way to handle this is to push the changes to composer.json and lock file as frequent as possible. Also I think we can suggest having 1 branch for all composer.json and lock file changes instead of mentioning `they should add that module in a separate branch`?

I think mostly both json and lock file changes together however I do not find any json file reference on above

Reply:
- I think composer.lock is the common issue
- Separate branch solution is also mentioned in the doc.

Reply:
Ok. Can we reword the 1st sentence as give above as we cannot avoid multiple people working on composer.lock in a large project with multiple developers

Reply:
@Yemaneberhan-Lemma this is good to be merged once above [comments](https://github.com/axelerant/engg-handbook/pull/72#discussion_r1270741813) are addressed

Reply:
@Kalaiselvan88 Done as per comment